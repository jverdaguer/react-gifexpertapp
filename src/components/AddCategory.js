import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const handleInput = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories((prev) => [inputValue, ...prev]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type='text' onChange={handleInput} value={inputValue} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
