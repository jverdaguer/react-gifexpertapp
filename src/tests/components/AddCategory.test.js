import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import { renderHook } from '@testing-library/react-hooks';

describe('Component <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('should Component exists <AddCategory />', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should change input box', () => {
        const input = wrapper.find('input');
        const value = 'Hello Word';
        input.simulate('change', { target: { value } });
        const inputAfter = wrapper.find('input');
        const newValue = inputAfter.prop('value');
        expect(value).toBe(newValue);
    });

    test('should not call setCategories on submit when category is empty', () => {
        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault() {} });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and inputValue should be empty', async () => {
        // 1. add a value to input box
        const input = wrapper.find('input');
        const value = 'Hello Word';
        input.simulate('change', { target: { value } }); // simulate e object

        // 2. Click on Submit
        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault() {} }); // e object needs a preventDefault method

        // 3. setCategories should be executed
        expect(setCategories).toHaveBeenCalled();

        // 4. Finally inputValue should be set to empty string
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});
