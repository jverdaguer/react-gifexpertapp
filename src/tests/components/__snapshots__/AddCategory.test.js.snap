// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <AddCategory /> should Component exists <AddCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
